account/recovery.go

```diff
@@ -523,3 +525,53 @@ func GenerateRecoveryKeys(ctx context.Context, accountTarget uint32,
 	}
 	return acctKeys, nil
 }
+
+// AdvanceAccountDerivationIndex asserts that the internal wallet has at least
+// the given minimum index for deriving account keys.
+func AdvanceAccountDerivationIndex(ctx context.Context, minimumIndex uint32,
+	wallet lndclient.WalletKitClient, chainParams *chaincfg.Params) error {
+
+	allAccounts, err := wallet.ListAccounts(
+		ctx, "", walletrpc.AddressType_UNKNOWN,
+	)
+	if err != nil {
+		return fmt.Errorf("error listing accounts: %v", err)
+	}
+
+	poolAccountsPath := fmt.Sprintf("m/%d'/%d'/%d'",
+		keychain.BIP0043Purpose, chainParams.HDCoinType,
+		poolscript.AccountKeyFamily)
+
+	var poolAccountsAccount *walletrpc.Account
+	for idx, acct := range allAccounts {
+		if acct.DerivationPath == poolAccountsPath {
+			poolAccountsAccount = allAccounts[idx]
+		}
+	}
+
+	// If we did find our account in the list and already have enough keys
+	// derived, we don't need to do anything here.
+	if poolAccountsAccount != nil &&
+		poolAccountsAccount.ExternalKeyCount > minimumIndex {
+
+		log.Debugf("Account %s already has %d external keys (want "+
+			"minimum index %d), not deriving any keys",
+			poolAccountsPath, poolAccountsAccount.ExternalKeyCount,
```

Comment:
I guess `wallet.ListAccounts` returns them in increasing order so we have the last one here? 

https://github.com/btcsuite/btcwallet/blob/50ea8d1125561279f9c9af1e46bea16ca414b4a5/wallet/wallet.go#L2396

Reply:
Yes, it returns them in order, but there are more than 255 and we want one towards the back (key family 220).